jaeger-influxdb/internal/queries.go

```diff
@@ -18,20 +18,12 @@ func traceIDToString(traceID model.TraceID) string {
 }
 
 func queryGetAllWhereTraceID(table string, traceIDs ...model.TraceID) string {
-	var whereClause string
-	if len(traceIDs) > 0 {
-		traceIDStrings := make([]string, len(traceIDs))
-		for i, traceID := range traceIDs {
-			traceIDStrings[i] = traceIDToString(traceID)
-		}
-		whereClause = fmt.Sprintf(`"%s" IN ('%s')`,
-			common.AttributeTraceID, strings.Join(traceIDStrings, `','`))
-	} else {
-		whereClause = "false"
+	traceIDStrings := make([]string, len(traceIDs))
+	for i, traceID := range traceIDs {
+		traceIDStrings[i] = traceIDToString(traceID)
 	}
-	query := fmt.Sprintf("SELECT * FROM %s WHERE %s",
-		table, whereClause)
-	return query
+	return fmt.Sprintf(`SELECT * FROM %s WHERE "%s" IN ('%s')`,
+		table, common.AttributeTraceID, strings.Join(traceIDStrings, `','`))
```

Comment:
presumably traceIDs won't be zero anymore (otherwise the in RHS will be empty.

Reply:
The traceIDs may be empty. If so, the where clause is `WHERE "trace_id" IN ('')`, which is valid, and returns zero results.

Of course, this assumes the trace_id column contains no empty string values, which I cannot guarantee.

Reply:
ah for string/char columns that works. I tested on an int column. 🤦